src/app/components/student/student.component.ts

```diff
@@ -103,4 +132,61 @@ export class StudentComponent implements OnInit {
       return 'asc';
     }
   }
+
+  filterTable(field: string, event): void {
+    this.filterSubject.next({field: field, value: event.target.value});
+  }
+
+  getFilterArray(newFilter: FilterModel, actualFilterArray: Array<FilterModel>): Array<FilterModel> {
+    const filterArray = [...actualFilterArray];
+    const fieldPredicate = (item: FilterModel) => item.field === newFilter.field;
+
+    if (filterArray.length === 0) {
+      if (newFilter.field && newFilter.value) {
+        filterArray.push(newFilter);
+        return filterArray;
+      } else {
+        return filterArray;
+      }
+    } else {
+      const existFilter = filterArray.find(fieldPredicate);
+
+      if (existFilter) {
+        const existIndex = filterArray.findIndex(fieldPredicate);
+
+        if (newFilter.field && newFilter.value) {
+          filterArray.splice(existIndex, 1);
+          filterArray.push(newFilter);
+        } else {
+          filterArray.splice(existIndex, 1);
+        }
+      } else {
+        filterArray.push(newFilter);
+      }
+
+      return filterArray;
+    }
+  }
+
+  getQuery(filterArray: Array<FilterModel>): string {
+    const filters: Array<string> = [];
+    filterArray.forEach(item => {
+      filters.push(`filter=${item.field} ct ${item.value}`);
+    });
+    
+    const query = filters.length ? `${filters.join('&')}` : '';
+
+    return query;
+  }
+
+  showInputButton(field: string): boolean {
+    const existFilter = this.currentFilterArray.find(filter => filter.field === field);
+
+    return !!(existFilter && existFilter.value);
+  }
+
+  deleteInputButton(field: string, filterInput: HTMLInputElement): void {
```

Comment:
I would advise to name 'action' functions starting with 'on'.
For example 'onClearFilter'.